### PR TITLE
[Fix-16793] WorkerGroupChangeNotifier may can not detect cluster's first time change

### LINKED ISSUE #16793: [Bug] [WorkerGroupChangeNotifier] work group listeners may can not be triggered when cluster changed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

1. when the `listeners` is empty, `detectWorkerGroupChanges()` may execute at least 1 time, so when next time `detectWorkerGroupChanges()` executing, `mapComparator.getValuesToAdd()` can not get the group differences again.

2. DEFAULT_REFRESH_WORKER_INTERVAL is too big, it will wait a long long time to detect again.

### What you expected to happen

when the `listeners` is empty, `mapComparator` should always keeps the differences.

### How to reproduce

create another worker group, start master-server

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by reele**
Fix #16793 
-- i created a new branch, the old is broken. @SbloodyS 

## Brief change log

1. add all groups to new listener when new listener added.

2. delete `DEFAULT_REFRESH_WORKER_INTERVAL` , add `startScheduleThread()` to get the parameter from `MasterConfig`.

3. add manual call `startScheduleThread()` in `ClusterManager.start()`

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java [hunk 1]
```diff
@@ -52,15 +55,23 @@ public class WorkerGroupChangeNotifier {
 
     public WorkerGroupChangeNotifier(WorkerGroupDao workerGroupDao) {
         this.workerGroupDao = workerGroupDao;
+    }
+
+    public void startScheduleThread() {
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java]:
```suggestion
    public void start() {
```

Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java]: Rename to `start` is better.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java [hunk 2]
```diff
@@ -52,15 +55,23 @@ public class WorkerGroupChangeNotifier {
...
+                workerGroupRefreshIntervalSeconds,
+                workerGroupRefreshIntervalSeconds,
                 TimeUnit.SECONDS);
     }
 
     public void subscribeWorkerGroupsChange(WorkerGroupListener listener) {
+
+        // add all group when listener added
+        listener.onWorkerGroupAdd(new ArrayList<>(workerGroupMap.values()));
+
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java]:
```suggestion
```
We don't need this line change? Sicne the listener should be triggered by `detectWorkerGroupChanges`

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/ClusterManager.java [hunk 3]
```diff
@@ -51,6 +51,7 @@ public void start() {
         this.registryClient.subscribe(RegistryNodeType.MASTER.getRegistryPath(), masterClusters);
         this.registryClient.subscribe(RegistryNodeType.WORKER.getRegistryPath(), workerClusters);
         this.workerGroupChangeNotifier.subscribeWorkerGroupsChange(workerClusters);
+        this.workerGroupChangeNotifier.startScheduleThread();
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/ClusterManager.java]:
```suggestion
        this.workerGroupChangeNotifier.start();
```

**Review by ruanwenjun — COMMENTED** [review]
Good catch, I left some comments.

**Comment by ruanwenjun** [comment]
Since we use a daemon thread to detect the worker group changed, so there will always exist delay, but this delay does have a serious effect. We can broadcast a worker group changed event to all masters once the api receive a worker group changed event.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java (hunk 2 above)
Comment (reele) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerGroupChangeNotifier.java]: yes, it's useless for currently

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Review by SbloodyS — COMMENTED** [review]
Please run `mvn spotless:apply` to format the code.

**Comment by reele** [comment]
formatted

**Review by SbloodyS — APPROVED** [review]
+1

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**